client/client.go

```diff
@@ -305,10 +299,13 @@ func (c *Client) downloadTimestamp() error {
 func (c *Client) downloadSnapshot() error {
 	logrus.Debug("downloadSnapshot")
 	role := data.RoleName("snapshot")
+	if c.local.Timestamp == nil {
```

Comment:
I added ErrMissingMeta to indicate we are missing the metadata for the role being downloaded. I can add a comment to the error definition to make that clear.

This is a check to stop somebody calling downloadSnapshot directly or out of order. I felt it was necessary so that we never call downloadSigned with a nil expectedSha256 when downloading a snapshot. The same reasoning applies to the checks in downloadTargets/GetTargetsFile